grakn-graph/src/main/java/ai/grakn/graph/internal/ThingImpl.java

```diff
@@ -251,19 +251,21 @@ protected T type(TypeImpl type) {
     /**
      *
      * @param type The type of this concept
-     * @return The concept itself casted to the correct interface
      */
-    private T setInternalType(Type type){
+    private void setInternalType(Type type){
         cachedInternalType.set(type.getLabel());
         vertex().property(Schema.VertexProperty.INSTANCE_TYPE_ID, type.getLabelId().getValue());
-        return getThis();
     }
 
     /**
      *
      * @return The id of the type of this concept. This is a shortcut used to prevent traversals.
      */
-    public Label getInternalType(){
+    Label getInternalType(){
         return cachedInternalType.get();
     }
+
+    public static ThingImpl<?, ?> from(Thing thing){
+        return (ThingImpl) thing;
+    }
```

Comment:
This one isn't always safe. I'd consider removing it and just casting. Unsafe example:
```
RelationImpl rel = thingy();
ThingImpl.from(rel); // crash
```